### DISCUSSION TITLE
WebAuthn as a first class citizen, like TOTP (config in User Account Service + optional execution in flows)

### DISCUSSION THREAD

**Discussion by awoimbee** [keycloak/keycloak#8810] [Ideas]
Here is an old issue about adding WebAuthn management in the User Account Service: https://issues.redhat.com/browse/KEYCLOAK-14298

Currently, only an admin can manage WebAuthn credentials, from the management console. This is an unnecessary workload for the admin and a hassle for users that have to contact him in order to update their credentials.
Adding WebAuthn management to the User Account Service would remove these pain points and is the logical next step in WebAuthn integration.

But this feature requires another change:
The User Account Service currently doesn't know if an authentication method is disabled (not present in any flow), thus it will always show the "Authenticator" section even if TOTP is disabled. We don't want to show a WebAuthn section if the default authentication flows don't use it.
So before Adding a "WebAuthn" section, the logical thing to do would be to add it as an optional `execution` in the default authentication `flow`s 

Thus I propose 3 tickets that must be done in order:
- Very near future: Add WebAuthn as an optional MFA login method in the default authentication flows
- Near future: Add a WebAuthn page in the User Account Service, equivalent to what the "Authenticator" page is for TOTP
- Distant future: User Account Service: hide credential providers that aren't part of any Flow (so aren't used)

**Comment by stianst** [comment]
The account console already shows the enabled authenticators and allow users to manage them. WebAuthN isn't enabled by default though as that is currently a preview feature since we're trying to find ways to automate testing around it. Going to close this issue as it's just duplicating the issues around making WebAuthN supported, which are currently in JIRA, but will be migrated to GH Issues soon.

**Comment by awoimbee** [comment]
> The account console already shows the enabled authenticators and allow users to manage them.

Wow, this is great ! I wasn't using the `keycloak.v2` theme, which  is necessary for this feature.

> WebAuthN isn't enabled by default though as that is currently a preview feature (...)

It is a preview feature, but it is enabled by default ([doc](https://www.keycloak.org/docs/latest/server_installation/index.html#profiles)

EDIT: Thanks for the awesome work on this project :)
